src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentParamNodeCodeFixProvider.cs

```diff
@@ -0,0 +1,161 @@
+                }
+                
+                var newNodeList = new SyntaxNode[]
+                {
+                    GetNewNode(name, isFirstNodeInComment: newDocComment.ChildNodes().First() == nodeAfterNewParamNode)
+                };
+
+                newDocComment = nodeBeforeNewParamNode != null
+                    ? newDocComment.InsertNodesAfter(nodeBeforeNewParamNode, newNodeList)
+                    : newDocComment.InsertNodesBefore(nodeAfterNewParamNode, newNodeList);
+            }
+
+            var newRoot = root.ReplaceNode(docCommentNode, newDocComment.WithAdditionalAnnotations(Formatter.Annotation));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private IEnumerable<TXmlElementSyntax> GetParamNodes(SyntaxNode docComment)
+        {
+            var paramNodes = docComment.ChildNodes().OfType<TXmlElementSyntax>()
+                                                    .Where(w => GetXmlElementLocalName(w) == "param");
+
+            // Prefer to place the doc comment in the outer level, as in auto-created comments
+            if (!paramNodes.Any())
+            {
+                paramNodes = docComment.DescendantNodes(descendIntoChildren: _ => true)
+                    .OfType<TXmlElementSyntax>()
+                    .Where(w => GetXmlElementLocalName(w) == "param");
+            }
+
+            return paramNodes;
+        }
+
+        private bool NodeExists(IEnumerable<TXmlElementSyntax> paramNodes, string name)
+        {
+            return paramNodes.Select(GetNameAttributes)
+                             .Where(nameAttributes => nameAttributes.Count == 1)
+                             .Any(nameAttributes => nameAttributes.Select(GetValueFromNameAttribute).Contains(name));
+        }
+
+        protected TXmlElementSyntax GetLastParamNodeCorrespondingToParamInList(IEnumerable<TXmlElementSyntax> paramNodeList, List<string> methodParamSubset)
+        {
+            TXmlElementSyntax nodeAfterNewParamNode = null;
+
+            foreach (var paramNode in paramNodeList)
+            {
+                var paramNameForNode = GetNameAttributes(paramNode);
+
+                // param node is missing `name` attribute or there are multiple `name` attributes
+                if (paramNameForNode.Count != 1)
+                {
+                    continue;
+                }
+                
+                if (methodParamSubset.Contains(GetValueFromNameAttribute(paramNameForNode.Single())))
+                {
+                    nodeAfterNewParamNode = paramNode;
+                }
+            }
+
+            return nodeAfterNewParamNode;
+        }
+
+        private class MyCodeAction : CodeAction.DocumentChangeAction
+        {
+            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument, string parameterName)
+                : base(string.Format(FeaturesResources.Add_missing_param_nodes, parameterName), createChangedDocument)
```

Comment:
"Add_missing_param_nodes" has no placeholders, so there should be no string.Format.